File: src/Api/Model/DB/CNAE/CnaeSubclasse.cs
using Newtonsoft.Json;
using Realms;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CriadorBaseDados.Model.DB.CNAE
{
    [JsonObject(MemberSerialization.OptIn)]
    /// <summary>
    /// Seção -> Divisão -> Grupo -> Classe -> Subclasse -> Atividade Econômica
    /// </summary>
    public class CnaeSubclasse : RealmObject
    {
        [JsonProperty]
        [PrimaryKey]
        public int ID { get; set; }
        [JsonProperty]
        [Indexed]
        public string Descricao { get; set; }
        [JsonProperty]
        public CnaeClasse Classe { get; set; }

        [JsonProperty]
        [Backlink(nameof(Empresa.CNAE_Fiscal))]
        public IQueryable<Empresa> Empresas { get;}
    }
}

File: src/Mobile/b2b/lib/model/core/cnae/cnae_subclasse.dart
import '../empresa.dart';

/// <summary>
/// Seção -> Divisão -> Grupo -> Classe -> Subclasse -> Atividade Econômica
/// </summary>
class CnaeSubclasse {
  final int id;
  final String descricao;
  final int codClasse;
  final List<Empresa> empresas;

  CnaeSubclasse(this.id, this.descricao, this.codClasse, this.empresas);

  CnaeSubclasse.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        descricao = json['descricao'],
        codClasse = json['codClasse'],
        empresas = json['empresas'] ?? List.empty(growable: true);

  Map<String, dynamic> toMap() => <String, dynamic>{
        'id': id,
        'descricao': descricao,
        'codClasse': codClasse,
        //'empresas': empresas,
      };
}
